### PR TITLE
MAX aggregation

### LINKED ISSUE #4: Add MAX aggregation

### PR DESCRIPTION
**Pull request by mcarmonaa**
Closes #4

### CODE REVIEW

File: sql/expression/aggregation.go [hunk 1]
```diff
@@ -107,3 +107,57 @@ func (e *First) Merge(buffer, partial sql.Row) {
...
 	return buffer[0]
 }
+
+// Max agregation returns the greatest value of the selected column.
+// It implements the AggregationExpression interface
+type Max struct {
+	UnaryExpression
+}
+
+func NewMax(e sql.Expression) *Max {
```
Comment (erizocosmico) [inline sql/expression/aggregation.go]: missing comments for all other exported members

Comment (mcarmonaa) [reply sql/expression/aggregation.go]: Done

File: sql/expression/aggregation_test.go [hunk 2]
```diff
@@ -96,3 +97,53 @@ func TestFirst_Eval(t *testing.T) {
...
 }
+
+func TestMax_Name(t *testing.T) {
+	assert := require.New(t)
+
+	m := NewMax(NewGetField(0, sql.Int32, "field", true))
+	assert.Equal("max(field)", m.Name())
+}
+
+func TestMax_Eval_Int32(t *testing.T) {
```
Comment (ajnavarro) [inline sql/expression/aggregation_test.go]: Can you add some tests with nulls involved? thew must be ignored. Also, check the corner cases on MAX here:

http://www-cs-students.stanford.edu/~wlam/compsci/sqlnulls

Comment (mcarmonaa) [reply sql/expression/aggregation_test.go]: @ajnavarro since the NULL thing is an enhancement to be implemented #33, can I merge this PR?

Comment (erizocosmico) [reply sql/expression/aggregation_test.go]: Even if nulls are not implemented per se, we can start handling nulls inside every node we implement.

**Comment by mcarmonaa** [comment]
@ajnavarro @erizocosmico Now `NULL` values are handled in the `MAX` node although when the #33 will be implemented, this logic could be useless.

**Comment by erizocosmico** [comment]
Rebase needed
